### PR TITLE
[ranges] Improve concept index

### LINKED ISSUE #3870: [ranges] has many concepts used without \libconcept
e.g.,
```
  template<input_or_output_iterator I, sentinel_for<I> S, subrange_kind K>
    inline constexpr bool enable_borrowed_range<subrange<I, S, K>> = true;
```

or

```
  template<@\libconcept{input_range}@ V, @\libconcept{indirect_unary_predicate}@<iterator_t<V>> Pred>
    requires view<V> && is_object_v<Pred>
  class filter_view;
```

What's the intended policy for the use of `\libconcept`?

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #3870

### CODE REVIEW

File: source/ranges.tex [hunk 1]
```diff
@@ -1368,7 +1368,7 @@
       return to_address(ranges::begin(@\exposid{derived}@()));
     }
     constexpr auto data() const
-      requires range<const D> && contiguous_iterator<iterator_t<const D>> {
+      requires @\libconcept{range}@<const D> && contiguous_iterator<iterator_t<const D>> {
```
Comment (johelegp) [inline source/ranges.tex:1371]:
```suggestion
      requires @\libconcept{range}@<const D> && @\libconcept{contiguous_iterator}<iterator_t<const D>> {
```
There's 3 uses of `contiguous_iterator` missing the change.

File: source/ranges.tex [hunk 2]
```diff
@@ -4552,15 +4552,15 @@
...
                indirect_unary_predicate<const Pred, iterator_t<const V>>
     { return ranges::begin(@\exposid{base_}@); }
 
     constexpr auto end() requires (!@\exposconcept{simple-view}@<V>)
     { return @\exposid{sentinel}@<false>(ranges::end(@\exposid{base_}@), addressof(*@\exposid{pred_}@)); }
 
     constexpr auto end() const
-      requires range<const V> &&
+      requires @\libconcept{range}@<const V> &&
                indirect_unary_predicate<const Pred, iterator_t<const V>>
```
Comment (johelegp) [inline source/ranges.tex:4564]:
```suggestion
               @\libconcept{indirect_unary_predicate}@<const Pred, iterator_t<const V>>
```
There's 2 `indirect_unary_predicate` in master needing this change.

File: source/ranges.tex [hunk 3]
```diff
@@ -4965,7 +4965,7 @@
 \begin{codeblock}
 namespace std::ranges {
   template<@\libconcept{input_range}@ V>
-    requires view<V> && @\libconcept{input_range}@<range_reference_t<V>> &&
+    requires @\libconcept{view}@<V> && @\libconcept{input_range}@<range_reference_t<V>> &&
              (is_reference_v<range_reference_t<V>> ||
               view<range_value_t<V>>)
```
Comment (johelegp) [inline source/ranges.tex:4970]:
```suggestion
              @\libconcept{view}<range_value_t<V>>)
```

File: source/ranges.tex [hunk 4]
```diff
@@ -5046,7 +5046,7 @@
...
```
Comment (johelegp) [inline source/ranges.tex:5051]:
```suggestion
              @\libconcept{view}@<range_value_t<V>>)
```

File: source/ranges.tex [hunk 5]
```diff
@@ -5357,7 +5357,7 @@
...
```
Comment (johelegp) [inline source/ranges.tex:5362]:
```suggestion
              @\libconcept{view}@<range_value_t<V>>)
```

File: source/ranges.tex [hunk 6]
```diff
@@ -5464,7 +5464,7 @@
     (remove_reference_t<R>::size() <= 1);
 
   template<@\libconcept{input_range}@ V, @\libconcept{forward_range}@ Pattern>
-    requires view<V> && view<Pattern> &&
+    requires @\libconcept{view}@<V> && view<Pattern> &&
```
Comment (johelegp) [inline source/ranges.tex:5467]:
```suggestion
    requires @\libconcept{view}@<V> && @\libconcept{view}@<Pattern> &&
```

File: source/ranges.tex [hunk 7]
```diff
@@ -5713,7 +5713,7 @@
 \begin{codeblock}
 namespace std::ranges {
   template<@\libconcept{input_range}@ V, @\libconcept{forward_range}@ Pattern>
-    requires view<V> && view<Pattern> &&
+    requires @\libconcept{view}@<V> && view<Pattern> &&
```
Comment (johelegp) [inline source/ranges.tex:5716]:
```suggestion
    requires @\libconcept{view}@<V> && @\libconcept{view}@<Pattern> &&
```

File: source/ranges.tex [hunk 8]
```diff
@@ -5781,7 +5781,7 @@
...
```
Comment (johelegp) [inline source/ranges.tex:5784]:
```suggestion
    requires @\libconcept{view}@<V> && @\libconcept{view}@<Pattern> &&
```

File: source/ranges.tex [hunk 9]
```diff
@@ -6713,7 +6713,7 @@
 \begin{codeblock}
 namespace std::ranges {
   template<input_range V, size_t N>
```
Comment (johelegp) [inline source/ranges.tex:6715]:
```suggestion
  template<@\libconcept{input_range}@ V, size_t N>
```

**Review by johelegp — CHANGES_REQUESTED** [review]
Mostly a few missing changes I noticed in the context that GitHub shows me.

File: source/ranges.tex (hunk 1 above)
Comment (jensmaurer) [reply source/ranges.tex:1371]: fixed

File: source/ranges.tex (hunk 2 above)
Comment (jensmaurer) [reply source/ranges.tex:4564]: fixed

File: source/ranges.tex (hunk 3 above)
Comment (jensmaurer) [reply source/ranges.tex:4970]: fixed

File: source/ranges.tex (hunk 4 above)
Comment (jensmaurer) [reply source/ranges.tex:5051]: fixed

File: source/ranges.tex (hunk 5 above)
Comment (jensmaurer) [reply source/ranges.tex:5362]: fixed

File: source/ranges.tex (hunk 6 above)
Comment (jensmaurer) [reply source/ranges.tex:5467]: fixed

File: source/ranges.tex (hunk 7 above)
Comment (jensmaurer) [reply source/ranges.tex:5716]: fixed

File: source/ranges.tex (hunk 8 above)
Comment (jensmaurer) [reply source/ranges.tex:5784]: fixed

File: source/ranges.tex (hunk 9 above)
Comment (jensmaurer) [reply source/ranges.tex]: fixed

**Comment by tkoeppe** [comment]
@jensmaurer: I take it this has no visible diff in the main document? If so, go ahead and merge. Thanks!

**Comment by jensmaurer** [comment]
Just confirmed again, no visible diff in the main document.

**Pull request merged**